src/components/Dashboard/PendingTxs/PendingTxsList/index.tsx

```diff
@@ -23,29 +23,21 @@ const PendingTxsList = ({ size = 5 }: { size?: number }): ReactElement | null =>
   const { address } = useSelector(currentSafe)
   const chainId = useSelector(currentChainId)
 
-  const queue = useQueueTransactions()
+  const queueTxns = useSelector(pendingTransactions)
 
   const queuedTxsToDisplay: Transaction[] = useMemo(() => {
-    let txs: Transaction[] = []
+    if (!queueTxns) return []
 
-    if (!queue) {
-      return txs
-    }
-
-    txs = Object.values(Object.values(queue))
-      // get the transactions from 'queue.next' and 'queue.queue'
-      .map((queue) => queue.transactions)
-      .flat()
-      // get the Transaction[] per nonce
-      .map((el) => el[1])
-      // select the first Transaction per nonce
-      .map((el) => el[0])
-      .slice(0, size)
-
-    return txs
-  }, [queue, size])
+    return (
+      Object.values(queueTxns.next)
+        .concat(Object.values(queueTxns.queued))
```

Comment:
It is cleaner to use this selector instead of the `useQueueTransactions` hook.
Same result but more readable.

Reply:
Exactly. `useQueueTransactions` does some dark magic wrt deep links, we need to clean that up eventually.